internal/sail/server/server.go

```diff
@@ -73,14 +98,23 @@ func (s SailServer) closeRoom(room *Room) {
 	room.Close()
 }
 
-func (s SailServer) startRoom(w http.ResponseWriter, req *http.Request) {
+func (s SailServer) connectRoom(w http.ResponseWriter, req *http.Request) {
 	conn, err := upgrader.Upgrade(w, req, nil)
 	if err != nil {
-		log.Printf("startRoom: %v", err)
+		log.Printf("connectRoom: %v", err)
+		return
+	}
+
+	roomID := req.URL.Query().Get(sailCommon.RoomIDKey)
+	secret := req.Header.Get(sailCommon.SecretKey)
+
+	room, err := s.getRoomWithAuth(RoomID(roomID), secret)
+	if err != nil {
+		log.Printf("connectRoom: %v", err)
```

Comment:
Do we want to set HTTP 500 here?

Reply:
ooh good call!

Reply:
...I don't care enough about parsing the errors to know if it should be 403 or 404 tho, so imma leave a todo